### PR TITLE
drivers: nrf: select PINCTRL

### PR DESCRIPTION
**Pull request by gmarull**

### CODE REVIEW

File: drivers/spi/Kconfig.nrfx [hunk 1]
```diff
@@ -6,6 +6,7 @@ menuconfig SPI_NRFX
 	default y
 	depends on SOC_FAMILY_NRF
 	depends on MULTITHREADING
+	select PINCTRL
```
Comment (nordicjm) [inline drivers/spi/Kconfig.nrfx:9]: qspi should select this also

Comment (nordicjm) [reply drivers/spi/Kconfig.nrfx:9]: AUDIO_DMIC_NRFX_PDM too

Comment (gmarull) [reply drivers/spi/Kconfig.nrfx:9]: done thanks

**Review by fabiobaltieri — APPROVED** [review]

**Review by rerickson1 — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]
QSPI NOR Kconfig looks ok.

**Pull request merged**